src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -723,6 +725,8 @@ private Path discoverSdkPath() {
           // prevent interference from exceptions in other resolvers
           logger.log(Level.SEVERE, resolver.getClass().getName()
               + ": exception thrown when searching for Google Cloud SDK", ex);
+          throw new CloudSdkNotFoundException(
```

Comment:
This breaks the outer `for` loop. I can we can recover from individual resolver exceptions and continue trying other resolvers. Also it doesn't seem right to throw the "not-found" exception when one of the resolvers just exploded.